src/main/java/at/hannibal2/skyhanni/features/inventory/MenuItemDisplayOverlayAbiphone.kt

```diff
@@ -0,0 +1,136 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.RenderItemTipEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.cleanName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class MenuItemDisplayOverlayAbiphone {
+    private val xOutOfYNoColorRequiredPattern = ".*: (§.)?(?<useful>[0-9]+)(§.)?\/(§.)?(?<total>[0-9]+).*".toPattern()
+
+    @SubscribeEvent
+    fun onRenderItemTip(event: RenderItemTipEvent) {
+        event.stackTip = getStackTip(event.stack)
+    }
+
+    private fun getStackTip(item: ItemStack): String {
+        if (SkyHanniMod.feature.inventory.menuItemNumberAbiphoneAsStackSize.isEmpty()) return ""
+        val itemName = item.cleanName()
+        val stackSizeConfig = SkyHanniMod.feature.inventory.menuItemNumberAbiphoneAsStackSize
+        val chestName = InventoryUtils.openInventoryName()
+        /*
+        -------------------------------IMPORTANT------------------------------------
+        
+        If at *any* point someone tells you to not nest your code, do the following
+        (in any order, but more importantly in the order given below):
+        - tell them to "cope and seethe", especially if they cite CodeAesthetic
+        - kindly remind them that the last time someone attempted this, friendships
+          were almost shattered in the process
+        - you will lose your sanity as you try to figure out what the fuck went wrong
+          in your forays with string manipulation
+        - remind them of the Single-responsibility Principle:
+          https://en.wikipedia.org/wiki/Single-responsibility_principle
+        - make sure you have an IDE capable of debugging within your reach
+        
+        This concludes the PSA. Happy writing! -Erymanthus
+
+        PS: T'was all a joke. Just don't do stupid shit like
+        ` if (!(chestName == "Visitor's Logbook")) return "" `
+        and you *should* be fine for the most part.
+        ----------------------------------------------------------------------------
+        */
+
+        //NOTE: IT'S String.length, NOT String.length()!
+        
+        if ((stackSizeConfig.contains(0)) && (chestName.contains("Abiphone") || chestName.contains("AⒷiphone")) && (itemName == ("Contacts Directory"))) {
```

Comment:
```
if ((stackSizeConfig.contains(0))
```

All those "magic numbers" make it hard to understand what is checked for

i suggest using an enum for those or constants, to understand what is checked better

Reply:
```
(chestName.contains("Abiphone") || chestName.contains("AⒷiphone"))
```

I suggest refactoring this into a method, as there are many places this is used and checked for, and if in the future hypixel adds more variants, it is easier to adapte

Reply:
> there are many places this is used

for now it's within the new files i created, but if it's any consolation i've condensed it into one boolean value within the offending file

as for adding it into lorenzutils, right now the one decent apiphone feature is in NEU and all i really added were stack sizes. we'll see though

Reply:
> ```
> if ((stackSizeConfig.contains(0))
> ```
> 
> All those "magic numbers" make it hard to understand what is checked for
> 
> i suggest using an enum for those or constants, to understand what is checked better

on the one hand, enums for readability would be nice
on the other hand, that's using a thermonuclear bomb to skin a cat (and anyone who knows how to use enums—aka not me–would probably be willing to match the magic numbers to the corresponding option in the config)